backend/app/controllers/eventsController/db/onTabChange.js

```diff
@@ -1,48 +1,47 @@
-    }
-
-    if (mongoEvent.operationType === "update") {
-
-        // Call waiter
-        if (mongoEvent.updateDescription.updatedFields?.callWaiter === 'called') {
-            console.log('  ');
-            const tab = await Tabs.findById(mongoEvent.documentKey._id)
-                .populate('tableId')
-                .exec();
-
-            const ownerRoomName = `${tab.restaurantId}:owner`;
-
-            io.to(ownerRoomName).emit("waiter:notification", { tableNo: tab.tableId.tableNo, tabId: tab._id });
-            console.log(`[Events] New waiter request sent in room: "${ownerRoomName}"`);
-        }
-
-        // Close tab
-        if (mongoEvent.updateDescription.updatedFields?.status === 'closed') {
-            console.log('  ');
-            const tab = await Tabs.findById(mongoEvent.documentKey._id);
-
-            const ownerRoomName = `${tab.restaurantId}:owner`;
-            const tabRoomName = `${tab.restaurantId}:${tab._id}`;
-
-            io.to(ownerRoomName).emit("tab:closed", { tableId: tab.tableId, tabId: tab._id });
-            console.log(`[Events] Tab closed sent in room: "${ownerRoomName}"`);
-
-            io.to(tabRoomName).emit("tab:closed", { tabId: tab._id });
-            console.log(`[Events] Tab closed sent in room: "${tabRoomName}"`);
-        }
-    }
+	if (mongoEvent.operationType === 'insert') {
+		console.log('Tab inserted');
+		// Give some time for the document to be created and propagated
+		setTimeout(() => {
+			const tab = mongoEvent.fullDocument;
+			const ownerRoomName = `${tab.restaurantId}:owner`;
+
+			io.to(ownerRoomName).emit('tab:open', { tableId: tab.tableId, tabId: tab._id });
+			console.log(`[Events] [Delayed ${propagationDelay/1000}s] New tab open request sent in room: "${ownerRoomName}"`);
+		}, propagationDelay);
+	}
+
+	if (mongoEvent.operationType === 'update') {
+
+		// Call waiter
+		if (mongoEvent.updateDescription.updatedFields?.callWaiter === 'called') {
+			console.log('  ');
+			const tab = await Tabs.findById(mongoEvent.documentKey._id)
+				.populate('tableId')
+				.exec();
+
+			const ownerRoomName = `${tab.restaurantId}:owner`;
+
+			io.to(ownerRoomName).emit('waiter:notification', { tableNo: tab.tableId.tableNo, tabId: tab._id });
+			console.log(`[Events] New waiter request sent in room: "${ownerRoomName}"`);
+		}
+
+		// Close tab
+		if (mongoEvent.updateDescription.updatedFields?.status === 'closed') {
+			console.log('  ');
+			const tab = await Tabs.findById(mongoEvent.documentKey._id);
+
+			const ownerRoomName = `${tab.restaurantId}:owner`;
+			const tabRoomName = `${tab.restaurantId}:${tab._id}`;
+
+			io.to(ownerRoomName).emit('tab:closed', { tableId: tab.tableId, tabId: tab._id });
+			console.log(`[Events] Tab closed sent in room: "${ownerRoomName}"`);
+
+			io.to(tabRoomName).emit('tab:closed', { tabId: tab._id });
+			console.log(`[Events] Tab closed sent in room: "${tabRoomName}"`);
+		}
+	}
 };
\ No newline at end of file
```

Comment:
The code patch seems to be functional with no syntax errors. However, there are a few improvements that could be made:

1. Code formatting - using consistent indentation and adding a newline at the end of the file would make the code more readable.

2. Use strict equality operator (===) when comparing with a string rather than the loose equality operator (==).

3. Add error handling - it's important to handle any error that may occur during the execution of the function.

4. Consider refactoring the code - depending on the size or complexity of the application, separating the different functionality within this event listener into separate methods could make the code more modular, easier to test, and maintain.